File: 251003/Kapinskiy/Discussion/src/main/java/by/kapinskiy/Discussion/models/Note.java
package by.kapinskiy.Discussion.models;


import lombok.Data;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.*;

import java.io.Serializable;


@Table("tbl_note")
@Data
public class Note implements Serializable {

    @PrimaryKeyClass
    @Data
    public static class NoteKey implements Serializable {
        @PrimaryKeyColumn(name = "country", type = PrimaryKeyType.PARTITIONED)
        private String country;

        @PrimaryKeyColumn(name = "id", ordinal = 0, type = PrimaryKeyType.CLUSTERED)
        private Long id;

        @PrimaryKeyColumn(name = "issueId", ordinal = 1, type = PrimaryKeyType.CLUSTERED)
        private Long issueId;

        public NoteKey(Long issueId) {
            this.issueId = issueId;
        }
    }


    @Column("content")
    private String content;

    @PrimaryKey
    private NoteKey key;

}

File: 251002/Nikonorova/labs3-5/discussion/Models/Note.cs
using Cassandra.Mapping.Attributes;


namespace discussion.Models
{
    [Table("tbl_notes", AllowFiltering = true)]
    public class Note
    {
        [PartitionKey] public string Country { get; set; } = string.Empty;
        [ClusteringKey(0)] public long ArticleId { get; set; }
        [ClusteringKey(1)] public long Id { get; set; }
        public string Content { get; set; } = string.Empty ;
    }
}
